### DISCUSSION TITLE
Socket is not disconnecting

### DISCUSSION THREAD

**Discussion by vimalmistry** [uNetworking/uWebSockets.js#595] [General]
Hi, I have question.

- If user connected to uws but when he disable internet or turn on airplane mode. not closing app. uws doesn't fire disconnect event.
- how to check app.publish method if it sent successfully or not? 

I am using flutter and its dart:io library for communicate to the server. Is anyone have some experience? What should I do?

**Comment by e3dio** [comment]
You should decrease your idleTimeout time if you need to know if client disconnected, only way to know for sure for that case is have server auto-send ping at idleTimeout and if no response it will close connection for you. Otherwise it will wait for client to respond and backpressure will build up until idleTimeout, or if you have closeOnBackpressureLimit set it will close

**Reply by vimalmistry** [reply]
> You should decrease your idleTimeout time if you need to know if client disconnected, only way to know for sure for that case is have server auto-send ping at idleTimeout and if no response it will close connection for you. Otherwise it will wait for client to respond and backpressure will build up until idleTimeout

So. If I set idletimeout to 5min. and ping is working. it will not close after 5 min? because I tried this. It is disconnecting after idle but connected to the ws. or I have to send anything from client also?

**Reply by e3dio** [reply]
> It is disconnecting after idle but connected to the ws

The only reason it will disconnect at idleTimeout is client did not respond to the auto-ping. The connection could be dead, or phone sleep etc. 5 minutes is a long idleTimeout if you want socket to disconnect soon after connection goes dead. If you want it to disconnect soon after dies you need short idleTimeout

**Reply by e3dio** [reply]
> I have to send anything from client also?

And you don't need to send anything to keep connection open, idleTimeout is when server checks on client with ping after client has been idle, if client responds to ping it resets time, if no response it closes connection. This is new behavior in v19, you can test this with server, phone and console.logs
